spec/unit/AndroidManifest.spec.js

```diff
@@ -0,0 +1,311 @@
+            it('should get the activity launch mode', () => {
+                expect(activity.getLaunchMode()).toBe(ACTIVITY_LAUNCH_MODE);
+            });
+
+            it('should set the activity launch mode', () => {
+                activity.setLaunchMode('standard');
+                expect(activity.getLaunchMode()).toBe('standard');
+            });
+
+            it('should remove the launch mode if set to empty', () => {
+                activity.setLaunchMode();
+                expect(activity.getLaunchMode()).toBe(undefined);
+            });
+        });
+    });
+
+    describe('minSdkVersion', () => {
+        it('should get minSdkVersion', () => {
+            expect(manifest.getMinSdkVersion()).toBe(MIN_SDK_VERSION);
+        });
+
+        it('should set minSdkVersion', () => {
+            manifest.setMinSdkVersion(1);
+            expect(manifest.getMinSdkVersion()).toBe(1);
+        });
+
+        it('should create the uses-sdk node if it does not exist when setting minSdkVersion', () => {
+            const NO_SDK_MANIFEST = DEFAULT_MANIFEST.replace(/<uses-sdk.*\/>/, '');
+            createTempManifestFile(NO_SDK_MANIFEST);
+            manifest = new AndroidManifest(manifestPath);
+
+            expect(manifest.doc.getroot().find('./uses-sdk')).toBe(null);
+
+            manifest.setMinSdkVersion(1);
+
+            expect(manifest.doc.getroot().find('./uses-sdk')).not.toBe(null);
+            expect(manifest.getMinSdkVersion()).toBe(1);
+        });
+    });
+
+    describe('maxSdkVersion', () => {
+        it('should get maxSdkVersion', () => {
+            expect(manifest.getMaxSdkVersion()).toBe(MAX_SDK_VERSION);
+        });
+
+        it('should set maxSdkVersion', () => {
+            manifest.setMaxSdkVersion(1);
+            expect(manifest.getMaxSdkVersion()).toBe(1);
+        });
+
+        it('should create the uses-sdk node if it does not exist when setting maxSdkVersion', () => {
+            const NO_SDK_MANIFEST = DEFAULT_MANIFEST.replace(/<uses-sdk.*\/>/, '');
+            createTempManifestFile(NO_SDK_MANIFEST);
+            manifest = new AndroidManifest(manifestPath);
+
+            expect(manifest.doc.getroot().find('./uses-sdk')).toBe(null);
+
+            manifest.setMaxSdkVersion(1);
+
+            expect(manifest.doc.getroot().find('./uses-sdk')).not.toBe(null);
+            expect(manifest.getMaxSdkVersion()).toBe(1);
+        });
+    });
+
+    describe('targetSdkVersion', () => {
+        it('should get targetSdkVersion', () => {
+            expect(manifest.getTargetSdkVersion()).toBe(TARGET_SDK_VERSION);
+        });
+
+        it('should set targetSdkVersion', () => {
+            manifest.setTargetSdkVersion(1);
+            expect(manifest.getTargetSdkVersion()).toBe(1);
+        });
+
+        it('should create the uses-sdk node if it does not exist when setting targetSdkVersion', () => {
+            const NO_SDK_MANIFEST = DEFAULT_MANIFEST.replace(/<uses-sdk.*\/>/, '');
```

Comment:
`/<uses-sdk.*?\/>/` or `/<uses-sdk[^>]*\/>/` will be safer if someone decides to add something after the last element.

Or you could not use a RegeExp to parse XML (kidding :wink:) and manipulate your doc after parsing:
```js
manifest.doc.getroot().remove(manifest.doc.getroot().find('./uses-sdk'))
```

Reply:
Ah, that's much better! I couldn't quite figure out the XML lib, and the docs didn't describe how to remove a node. I'll do this, thanks!